libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -103,28 +108,60 @@ internal class MapboxTripSession(
                         roadObjects = getRouteInitInfo(it)?.roadObjects ?: emptyList()
                     }
                     this@MapboxTripSession.primaryRoute = newPrimaryRoute
+                    logD("primary route update - finished", LOG_CATEGORY)
                 }
-                threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
-                    navigator.setAlternativeRoutes(routes.drop(1))
-                }
+                val updateAlternativesJob =
+                    threadController.getMainScopeAndRootJob().scope.launch(
+                        Dispatchers.Main.immediate
+                    ) {
+                        logD("alternative routes update - starting", LOG_CATEGORY)
+                        processedAlternatives.addAll(
+                            navigator.setAlternativeRoutes(routes.drop(1))
+                        )
+                        logD("alternative routes update - finished", LOG_CATEGORY)
```

Comment:
so, does it mean that NN is ready to process alternatives before it finished processing a primary route?

Reply:
oh, we already do this.

Reply:
Sorry for double checking, but @DenisPryt can you please confirm that NN is ready to update alternatives while primary route update is in progress?

Reply:
This is correct. Sequence of method calls will be saved even without waiting for complete.

Reply:
@DenisPryt , just curious, does it mean that under the hood `setAlternativeRoutes` isn't processed until primary route is set?